src/main/java/io/gsonfire/GsonFireBuilder.java

```diff
@@ -230,7 +230,10 @@ public GsonBuilder createGsonBuilder(){
         }
 
         builder.registerTypeAdapterFactory(new SimpleIterableTypeAdapterFactory());
-        builder.registerTypeAdapterFactory(new WrapTypeAdapterFactory(wrappedClasses));
+
+        for(Map.Entry<Class, Mapper> wrappedEntries: wrappedClasses.entrySet()) {
+            builder.registerTypeAdapterFactory(new WrapTypeAdapterFactory(wrappedEntries.getKey(), wrappedEntries.getValue()));
```

Comment:
My only concern here is that if you register two wrappers, one for `Animal` and another one for `Cat`, you should be aware that it will wrap with the first registered. Apart from that, looks good to me.